src/components/UserProfile/FoodTicket.tsx

```diff
@@ -0,0 +1,171 @@
+"use client";
+
+import { generateClient } from "aws-amplify/api";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { QRCode } from "react-qrcode-logo";
+
+import { type Schema } from "@/amplify/data/resource";
+import { createMessageAndCode } from "@/amplify/function/BusinessLogic/utils/crytography";
+import { getUpcomingFoodEventDetails } from "@/app/get-food-ticket/actions";
+import { useUser } from "@/components/contexts/UserContext";
+import { useQuery } from "@tanstack/react-query";
+
+import ProfileLinks from "./ProfileLinks";
+
+export default function UserFoodTicket() {
+  const client = generateClient<Schema>();
+
+  const userId = useUser().currentUser.userSub as string;
+
+  const { data, isFetching } = useQuery({
+    initialDataUpdatedAt: 0,
+    queryKey: ["User", userId],
+    queryFn: async () => {
+      const {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+      } = await getUpcomingFoodEventDetails(userId);
+
+      let verificationCode = "";
+
+      const { data, errors } = await client.queries.GetUserMessageCode({
+        userMessage: userId,
+      });
+      if (errors) {
+        console.log(errors);
+        verificationCode = "Error when searching for your message code";
+        return;
+      }
+
+      if (data) {
+        const json = JSON.parse(data.body as string);
+        const code = json["value"];
+
+        const createdVerificationCode = createMessageAndCode(userId, code);
+        verificationCode = createdVerificationCode;
+      } else {
+        verificationCode = "Error when searching for your message code";
+      }
+
+      return {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+        verificationCode,
+      };
+    },
+  });
+
+  // Not sure if we actually plan on showing their use code and i have no idea what its used for so for container purposes i will shorten it
+  const shortenString = (str: string, maxLen: number) => {
+    if (str) {
+      if (str.length <= maxLen) {
+        return str;
+      }
+      return str.substring(0, maxLen) + "...";
+    }
+  };
+
+  const [foodTicketData, setFoodTicketData] = useState({
+    queuePosition: data?.queuePosition,
+    eventName: data?.eventName,
+    eventDescription: data?.eventDescription,
+    eventTime: data?.eventTime,
+    timeslot: data?.timeslot,
+    userCode: shortenString(data?.verificationCode as string, 15),
+  });
+
+  // Not sure if this is needed, but when queue becomes lower, than it should update
+  useEffect(() => {
+    let queuePosition;
+    if (data?.queuePosition === "0") {
+      queuePosition = "You are first in line!";
+    } else {
+      queuePosition = data?.queuePosition;
```

Comment:
Since queuePosition is just a number, you should show more than just a number, to help the person understand what that means. Like: you are in position 2 out of 3 for food